src/provider/model/Model.ts

```diff
@@ -0,0 +1,119 @@
+import {AppWindow} from "./AppWindow";
+import {ContextChannel} from "./ContextChannel";
+import {injectable, inject} from "inversify";
+import {Application} from "../../client/main";
+import {Inject} from "../common/Injectables";
+import {Environment} from "./Environment";
+import {Identity} from "openfin/_v2/main";
+import {AppDirectory} from "./AppDirectory";
+
+export enum FindFilter {
+    WITH_CONTEXT_LISTENER,
+    WITH_INTENT_LISTENER
+}
+
+export interface FindOptions {
+    prefer?: FindFilter;
+    require?: FindFilter;
+}
+
+@injectable()
+export class Model {
+    @inject(Inject.APP_DIRECTORY)
+    private _directory!: AppDirectory;
+
+    @inject(Inject.ENVIRONMENT)
+    private _environment!: Environment;
+
+    private _windows: AppWindow[];
+    private _channels: ContextChannel[];
+
+    constructor(@inject(Inject.ENVIRONMENT) environment: Environment) {
```

Comment:
Any way to avoid needing to inject in both the constructor and the class declaration?

Reply:
Could remove the `@inject` from the member and assign in constructor, or make this extend `AsyncInit` and add the listeners post-injection.

Any thoughts/preference?..

Reply:
Let's decide in that 60 second huddle!